datumaro/plugins/synthia_format.py

```diff
@@ -0,0 +1,180 @@
+    if label_map is None:
+        label_map = SYNTHIA_LABEL_MAP
+
+    categories = {}
+    label_categories = LabelCategories()
+    for label in label_map:
+        label_categories.add(label)
+    categories[AnnotationType.label] = label_categories
+
+    has_colors = any(v is not None for v in label_map.values())
+    if not has_colors: # generate new colors
+        colormap = generate_colormap(len(label_map))
+    else: # only copy defined colors
+        colormap = { label_id: (desc[0], desc[1], desc[2])
+            for label_id, desc in enumerate(label_map.values()) }
+    mask_categories = MaskCategories(colormap)
+    mask_categories.inverse_colormap # pylint: disable=pointless-statement
+    categories[AnnotationType.mask] = mask_categories
+    return categories
+
+def parse_label_map(path):
+    label_map = OrderedDict()
+    with open(path, 'r', encoding='utf-8') as f:
+        for line in f:
+            # skip empty and commented lines
+            line = line.strip()
+            if not line or line[0] == '#':
+                continue
+
+            # color, name
+            label_desc = line.split()
+
+            if 2 < len(label_desc):
+                name = label_desc[3]
+                color = tuple([int(c) for c in label_desc[:3]])
+            else:
+                name = label_desc[0]
+                color = None
+
+            if name in label_map:
+                raise ValueError("Label '%s' is already defined" % name)
+
+            label_map[name] = color
+    return label_map
+
+class SynthiaExtractor(SourceExtractor):
+    def __init__(self, path):
+        if not osp.isdir(path):
+            raise FileNotFoundError("Can't read dataset directory '%s'" % path)
+
+        super().__init__()
+
+        self._categories = self._load_categories(path)
+        self._items = list(self._load_items(path).values())
+
+    def _load_categories(self, path):
+        label_map_path = osp.join(path, SynthiaPath.LABELMAP_FILE)
+        if osp.isfile(label_map_path):
+            label_map = parse_label_map(label_map_path)
+        else:
+            label_map = SYNTHIA_LABEL_MAP
+        return make_categories(label_map)
+
+    def _load_items(self, root_dir):
+        image_dir = osp.join(root_dir, SynthiaPath.IMAGES_DIR)
+        if osp.isdir(image_dir):
+            images = {
+                osp.splitext(osp.relpath(p, image_dir))[0].replace('\\', '/'): p
+                for p in find_images(image_dir, recursive=True)
+            }
+        else:
+            images = {}
+
+        items = {}
+
+        inst_dir = osp.join(root_dir, SynthiaPath.LABELS_SEGM_DIR)
+        if osp.isdir(inst_dir):
+            gt_images = find_images(inst_dir, recursive=True)
+            for gt_img in gt_images:
+                item_id = osp.splitext(osp.relpath(gt_img, inst_dir))[0].replace('\\', '/')
+
+                anno = []
+                instances_mask = load_image(gt_img, dtype=np.uint16)
```

Comment:
Judging by how this is used, it doesn't really contain instance information, so it would be better to use a different name.

Reply:
Fixed